### PR TITLE
Fix coveralls

### PR DESCRIPTION
**Pull request by LSchueler**
We are having problems with coveralls.io at the moment. By only running coveralls once during the `source_check` job, things should be running fine again.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install -v --editable .[lint]
```
Comment (MuellerSeb) [inline .github/workflows/main.yml:33]:
```suggestion
          pip install -v --editable .[lint, test]
```

Comment (MuellerSeb) [reply .github/workflows/main.yml:33]: To push coverage we also need to run tests, right?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -48,9 +49,15 @@ jobs:
...
-      #- name: cython-lint check
-        #run: |
-          #cython-lint src/gstools/
+      - name: cython-lint check
+        run: |
+          cython-lint src/gstools/
+
+      - name: coveralls check
+        run: |
+          python -m coveralls --service=github
```
Comment (MuellerSeb) [inline .github/workflows/main.yml:58]:
```suggestion
          python -m pytest --cov gstools --cov-report term-missing -v tests/
          python -m coveralls --service=github
```

Comment (MuellerSeb) [reply .github/workflows/main.yml:58]: I think we need to run pytest also.

File: src/gstools/field/summator.pyx [hunk 3]
```diff
@@ -10,7 +10,7 @@ if OPENMP:
     cimport openmp
 
 cimport numpy as np
-from libc.math cimport cos, pi, sin, sqrt
+from libc.math cimport cos, sin
```
Comment (MuellerSeb) [inline src/gstools/field/summator.pyx:13]: Ah, linting helps

**Review by MuellerSeb — CHANGES_REQUESTED** [review]
I think we need to run pytest as well.

Also, why is the index/readme hack in here?

File: .github/workflows/main.yml (hunk 1 above)
Comment (LSchueler) [reply .github/workflows/main.yml]: True, thanks! I forgot how coveralls.io works...

Comment (LSchueler) [reply .github/workflows/main.yml]: The syntax seems a bit strange, if you want to install multiple extra dependencies.

File: src/gstools/field/summator.pyx (hunk 3 above)
Comment (LSchueler) [reply src/gstools/field/summator.pyx:13]: It does!

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -30,7 +30,8 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install -v --editable .[lint]
+          pip install -v --editable '.[lint, test]'
```
Comment (MuellerSeb) [inline .github/workflows/main.yml:33]:
```suggestion
          pip install -v --editable .[lint,test]
```

Comment (MuellerSeb) [reply .github/workflows/main.yml:33]: I guess the whitespace is the issue

**Comment by LSchueler** [comment]
I think this is ready to be merged.

**Comment by MuellerSeb** [comment]
Please commit my suggestion, then its ready from my side.

File: .github/workflows/main.yml (hunk 4 above)
Comment (LSchueler) [reply .github/workflows/main.yml:33]: I checked again and the syntax causes problems in some shells, e.g. zsh. where `pip install -v --editable .[lint,test]` also doesn't work without the whitespace. I think I don't like whitespace dependent shell commands :-) So, I would just leave the apostrophes, but I can get rid of the whitespace.

**Pull request merged**